### PR TITLE
Issue/1/sims module stubs

### PR DESCRIPTION
**Pull request by akumgill**

### CODE REVIEW

**Review by cavestruz — CHANGES_REQUESTED** [review]

**Comment by akumgill** [comment]
requested changes addressed:

- when generating the population, we read from a sim_config and write output files back to the same folder the sim_config is in
- when doing inference, we read the sim_dir which has the output intermediate files from that step and we read the config_dir, from which we get our config params and to which we right the output of inference
- added subselection of mc_pairs based on some user specified cuts we want to make. currently the only option is to include all mc_pairs, but we can change that as we see fit

File: configs/example/infer_config.json
Comment (cavestruz) [inline configs/example/infer_config.json:1]: For this example config, I noticed that the inference type is sbi, but there are mcmc specific other kwargs, e.g. number of walkers, burn in, etc.  this, I think, should just information to grab a trained sbi inferrer, and/or create it, yes?

File: simulations/example/sim_config.json
Comment (cavestruz) [inline simulations/example/sim_config.json:1]: Quick requests - the short names require a bit of digging into the examples notebook `simulate_wl_profile.ipynb.`

- [ ] can we change `num_samples` to `num_parameter_samples` -- I first thought this was number of simulated clusters?
- [ ] can we change `num_drawn` to `num_sims` (As, this is number of simulations for training, yes?)

**Review by cavestruz — CHANGES_REQUESTED** [review]
Minor changes requested (mostly for naming clarity in the config files).  But, this looks great!

File: configs/example/infer_config.json
Comment (akumgill) [reply configs/example/infer_config.json:1]: yes, i have these params passed in to the infer config, but they aren't read/used if we're using SBI. i can add this just to MCMC implementation, but wasn't sure if these were params we'd want to tweak in config. no strong preference either way from my side

**Comment by akumgill** [comment]
requested changes addressed

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+python examples/sim_config_example.py
+
+Next, generate the infer_config file in the directory of interest:
+python examples/infer_config_example.py
+
+Then, use this sim_config to run the numerical experiment in the sim_dir folder:
+python examples/simulate_wl_profile.py --sim_dir example
+
+Next, run inference using the sim_config in sim_dir and the infer_config in config_dir. This will output to the infer_config (so if you want to run the same inference with different simulations, you should specify multiple infer_configs for that).
+python examples/mcmc_example.py --config_dir example --sim_dir example
```
Comment (akumgill) [inline README.rst:27]: `python examples/mcmc_example.py --infer_dir example --sim_dir example  --obs_dir example`

* Infer dir contains infer_config saying how to do inference (e.g. priors, walkers, burnin, etc) and is the output directory for inference
* Sim dir contains sim_config saying how a given set of simulations in that directory were made
* Obs dir contains obs_config saying how a given set of simulated/real observations were made/obtained

**Comment by akumgill** [comment]
@cavestruz bump

**Review by cavestruz — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**